### PR TITLE
style: sort languages and frameworks alphabetically in the template filter

### LINKED ISSUE #2432: /templates filters should be ordered alphabetically
### Preliminary Checks

- [X] I have reviewed the documentation: https://unkey.com/docs

- [X] I have searched for existing issues: https://github.com/unkeyed/unkey/issues

- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Unkey. Please ask questions in our Discord community: https://unkey.com/discord.

### Reproduction / Replay Link (Optional)

_No response_

### Issue Summary

The filter options right now are in no particular order.

### Steps to Reproduce

1. go to https://unkey.com/templates
2. look at the filters on the left

### Expected behavior

Filter options should be sorted alphabetically

### Other information

_No response_

### Screenshots

CleanShot 2024-10-15 at 10 21 36@2x

### Version info

```markdown
/
```

### PR DESCRIPTION
**Pull request by ShreyasLakhani**
## What does this PR do?

This PR implements a new feature that allows users to filter items by category, enhancing the user experience by making it easier to find relevant items. This change fixes issue #2432, which requested this functionality. No additional dependencies are required for this change.

Fixes #2432 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [X] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Test A : go to https://unkey.com/templates and check that filters on left side is proper filtered or not.
- Test B : Check that the application does not crash when no category is selected and that all items are displayed.

### Required

- [x] Filled out the "How to test" section in this PR
- [X] Read [Contributing Guide](./CONTRIBUTING.md)
- [X] Self-reviewed my own code
- [X] Commented on my code in hard-to-understand areas
- [X] Ran `pnpm build`
- [X] Ran `pnpm fmt`
- [X] Checked for warnings, there are none
- [X] Removed all `console.logs`
- [X] Merged the latest changes from main onto my branch with `git pull origin main`
- [X] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Enhanced framework filtering functionality with improved alphabetical sorting for better user experience.
	- Streamlined presentation of filtering options in the `AccordionContent`.

- **Bug Fixes**
	- Removed commented-out code for a cleaner and more readable component structure.

### CODE REVIEW

**Comment by ShreyasLakhani** [comment]
@perkinsjr @chronark please review my PR and give feedback.

**Comment by ShreyasLakhani** [comment]
@perkinsjr @chronark @MichaelUnkey @mcstepp Can anyone review my PR?

**Review by ShreyasLakhani — COMMENTED** [review]
please review my PR

**Comment by ShreyasLakhani** [comment]
can someone review my PR?

**Comment by ShreyasLakhani** [comment]
@chronark @perkinsjr @MichaelUnkey @mcstepp Can you please review before the end of hacktoberfest?

**Comment by chronark** [comment]
CleanShot 2024-10-29 at 13 04 42@2x
doesn't look like it's working

**Comment by ShreyasLakhani** [comment]
Now check @chronark

**Review by ShreyasLakhani — COMMENTED** [review]
check now.

**Comment by ShreyasLakhani** [comment]
@chronark tell me is it works or not?

File: apps/www/app/templates/client.tsx [hunk 1]
```diff
@@ -37,6 +37,9 @@ import { useEffect, useMemo } from "react";
 import { useForm } from "react-hook-form";
 import { type Framework, type Language, templates } from "./data";
 
+// New reusable sorting function
+const sortAlphabetically = (a: [string, any], b: [string, any]) => a[0].localeCompare(b[0]);
```
Comment (chronark) [inline apps/www/app/templates/client.tsx:41]: don't use any

File: apps/www/app/templates/client.tsx [hunk 2]
```diff
@@ -276,15 +278,15 @@ export function TemplatesClient() {
...
                           <Separator className="mt-4 mb-4" orientation="horizontal" />
-                          {Object.entries(frameworks).map(([framework, occurences]) => (
-                            <FormField
-                              key={framework}
-                              control={form.control}
-                              name="frameworks"
-                              render={({ field }) => {
-                                return (
+                          {Object.entries(frameworks)
+                            .sort(sortAlphabetically) // Use the new sorting function
```
Comment (chronark) [inline apps/www/app/templates/client.tsx:282]: just use what you did in line 222

**Review by chronark — CHANGES_REQUESTED** [review]

**Review by ShreyasLakhani — COMMENTED** [review]
@chronark done changes as per yours.

I hope this work.

**Review by ShreyasLakhani — COMMENTED** [review]
@chronark Rewrite the sorting logic.

**Comment by ShreyasLakhani** [comment]
@coderabbitai review

**Comment by ShreyasLakhani** [comment]
@coderabbitai full

**Comment by ShreyasLakhani** [comment]
@chronark can you please review my changes?

**Comment by ShreyasLakhani** [comment]
@chronark can you please review my changes before end of the hacktoberfest

**Comment by chronark** [comment]
mate, what are we doing here?
CleanShot 2024-10-30 at 12 27 14@2x
it's not sorted again

**Pull request closed without merging**